packages/fury-adapter-oas3-parser/lib/parser/oas/parseMediaTypeObject.js

```diff
@@ -76,13 +82,27 @@ const createJSONMessageBodyAsset = R.curry((namespace, mediaType, value) => {
   return asset;
 });
 
+const createTextMessageBodyAsset = R.curry((namespace, mediaType, value) => {
+  const asset = new namespace.elements.Asset(value.toValue());
+  asset.classes.push('messageBody');
+  asset.contentType = mediaType;
+  return asset;
+});
+
 function parseExample(namespace, mediaType) {
   const createExampleNotJSONWarning = createWarning(namespace,
-    "'Media Type Object' 'example' is only supported for JSON media types");
+    `'${name}' 'example' is not supported for media type '${mediaType}'`);
+
+  const isTextBasedType = R.anyPass([
+    () => isTextMediaType(mediaType),
+    () => isXMLMediaType(mediaType),
+  ]);
 
-  return R.ifElse(() => isJSONMediaType(mediaType),
-    R.compose(createJSONMessageBodyAsset(namespace, mediaType), getValue),
-    createExampleNotJSONWarning);
+  return R.compose(R.cond([
+    [() => isJSONMediaType(mediaType), createJSONMessageBodyAsset(namespace, mediaType)],
+    [() => isTextBasedType(mediaType), createTextMessageBodyAsset(namespace, mediaType)],
```

Comment:
Why is the extra closure needed? Simple `isJSONMediaType(...)` should be enough.

Reply:
It's the same as it was before `return R.ifElse(() => isJSONMediaType(mediaType),`. Otherwise the provided value to isJSONMediaType will be the example element and not the media type.

Reply:
So, you are saying the following line should be changed:

https://github.com/apiaryio/api-elements.js/blob/master/packages/fury-adapter-oas3-parser/lib/parser/oas/parseParameterObject.js#L94

```
[() => hasKey('name'), parseName],
```

Reply:
I am just wondering why the 2 cases are different. As far as I understand, they are not.

Reply:
`[() => hasKey('name'), parseName],` <- hasKey here takes two arguments, it is a curried function taking a string (key name) and a member element, it returns boolean. This proposed change wouldn't work and cause tests to fail.

isTextBasedType takes a single argument, the media type name.

Reply:
Yup, that was what I was missing in this context. Thanks.